src/auth/authContext.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import firestore, { auth } from "../firebaseConfig";
+
+export const AuthContext = React.createContext({
+  user: null,
+  setUser: null,
+});
+
+const AuthProvider = (props) => {
+  const [user, setUser] = useState();
+
+  useEffect(() => {
+    auth.onAuthStateChanged((auth) => {
+      processAuth(auth, setUser);
+    });
+  }, []);
+
+  return (
+    <AuthContext.Provider value={{ user, setUser }}>
+      {props.children}
+    </AuthContext.Provider>
+  );
+};
+
+export const processAuth = async (auth, setUser) => {
+  if (!auth) {
+    return setUser(null);
+  }
+  let firestoreResult = await firestore
+    .doc("users/" + auth.uid) // A unique user ID, assigned to the requesting user, we're calling/maticing this ID from our server.
```

Comment:
Thanks for adding comment about uid. You may write it above the code, rather than writing the comment same line with the code.